ablySpec/Auth.swift

```diff
@@ -599,5 +599,164 @@ class Auth : QuickSpec {
                 expect(createTokenRequestMethodWasCalled).to(beTrue())
             }
         }
+
+        // RSA9
+        describe("createRequestToken") {
+
+            // RSA9a
+            it("should create and sign a TokenRequest") {
+                let options = AblyTests.commonAppSetup()
+                let rest = ARTRest(options: options)
+                let expectedClientId = "client_string"
+                let tokenParams = ARTAuthTokenParams(clientId: expectedClientId)
+
+                rest.auth.createTokenRequest(tokenParams, options: options, callback: { tokenRequest, error in
+                    expect(tokenRequest?.clientId).to(equal(expectedClientId))
+                    expect(tokenRequest?.mac).toNot(beNil())
+                    expect(tokenRequest?.nonce).toNot(beNil())
+                })
+            }
+
+            // RSA9b
+            it("should support AuthOptions") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+                let auth: ARTAuth = rest.auth
+
+                let authOptions = ARTAuthOptions(key: "key:secret")
+
+                auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                    expect(tokenRequest?.keyName).to(equal("key"))
+                })
+            }
+
+            // RSA9c
+            it("should generate a unique 16+ character nonce if none is provided") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
```

Comment:
To test uniqueness please create at least least two token request objects and ensure the nonce is not the same